MeshLib/MeshGenerators/MeshLayerMapper.cpp

```diff
@@ -60,6 +61,10 @@ MeshLib::Mesh* MeshLayerMapper::createStaticLayers(MeshLib::Mesh const& mesh, st
 	std::vector<MeshLib::Node*> new_nodes(nNodes + (nLayers * nNodes));
 	std::vector<MeshLib::Element*> new_elems;
 	new_elems.reserve(nElems * nLayers);
+	MeshLib::Properties properties;
+	boost::optional<PropertyVector<int> &> materials = 
+		properties.createNewPropertyVector<int>("MaterialIDs", MeshLib::MeshItemType::Cell);
+	materials->reserve(nElems * nLayers);
```

Comment:
I do not mean the product of `nElems` and `nLayers` being 0; The `materials` variable of type `boost::optional<...>` can be empty. This must be checked before using pointer access for reserve call.